holonote/annotate/annotator.py

```diff
@@ -195,72 +190,127 @@ def selected_index(self):
+                    "Only 'range' region allowed for 'set_range', "
+                    f"{kdims[1]!r} is {r!r}."
+                )
+            regions = {kdims[0]: (value[0], value[1]), kdims[1]: (value[2], value[3])}
+
+        self.set_regions(**regions)
 
     def set_point(self, posx, posy=None):
-        if 'Point' not in self.region_types:
-            raise ValueError(f'Point region types not enabled as region_types={self.region_types}')
-
-        value = (posx,None) if posy is None else (posx, posy)
-        kdims = list(self.kdim_dtypes.keys())
-        self._set_region('Point', value,  *kdims)
-
-
-    def _set_region(self, region_type, value=None, dim1=None, dim2=None):
-        "Use _set_region(None) to clear currently defined region"
-        if (region_type, value, dim1, dim2) == (None, None, None, None):
-            self._region = {}
-        elif None not in [value, dim1]:
-            self._region = {'region_type':region_type, 'value':value, 'dim1':dim1, 'dim2':dim2}
+        print("set_point is legacy use set_regions instead")
+
+        kdims = list(self.spec)
+        if posy is None:
+            if len(kdims) != 1:
+                raise ValueError('Only one key dimension is allowed in spec.')
+            if (r := self.spec[kdims[0]]['region']) != 'single':
+                raise ValueError(
+                    "Only 'single' region allowed for 'set_point', "
+                    f"{kdims[0]!r} is {r!r}."
+                )
+            regions = {kdims[0]: posx}
         else:
-            raise Exception('Both value and dim1 required for non-None region type')
-
-    def add_annotation(self, **fields):  #   Rename box to range.
-        "Primary key specification is optional"
+            if len(kdims) != 2:
+                raise ValueError('Only two key dimensions is allowed in spec.')
+            if (r := self.spec[kdims[0]]['region']) != 'single':
+                raise ValueError(
+                    "Only 'single' region allowed for 'set_point', "
+                    f"{kdims[0]!r} is {r!r}."
+                )
+            if (r := self.spec[kdims[1]]['region']) != 'single':
+                raise ValueError(
+                    "Only 'single' region allowed for 'set_point', "
+                    f"{kdims[1]!r} is {r!r}."
+                )
+            regions = {kdims[0]: posx, kdims[1]: posy}
+
+        self.set_regions(**regions)
+
+    def _add_annotation(self, **fields):
+        # Primary key specification is optional
         if self.connector.primary_key.field_name not in fields:
             index_val = self.connector.primary_key(self.connector,
                                                    list(self.annotation_table._field_df.index))
             fields[self.connector.primary_key.field_name] = index_val
 
-        if self.region != self._last_region:
+        # Don't do anything if self.region is an empty dict
+        if self.region and self.region != self._last_region:
```

Comment:
Shouldn't the line below should be removed?

```
if len(self.annotation_table._annotators)>1:
```

There can only be one annotator now...

Reply:
Agree. I will clean it up in another PR. 

This logic is still there to keep changes to a minimum.